apps/ios/Shared/Views/UserSettings/UserProfilesView.swift

```diff
@@ -41,45 +54,103 @@ struct UserProfilesView: View {
+                        Label("Add profile", systemImage: "plus")
+                    }
+                    .frame(height: 44)
+                    .padding(.vertical, 4)
                 }
-                .frame(height: 44)
-                .padding(.vertical, 4)
             } footer: {
-                Text("Your chat profiles are stored locally, only on your device.")
+                Text("Tap to activate profile.")
+                    .font(.body)
+                    .padding(.top, 8)
+
             }
         }
         .toolbar { EditButton() }
         .navigationTitle("Your chat profiles")
+        .searchable(text: $searchTextOrPassword, placement: .navigationBarDrawer(displayMode: .always))
+        .autocorrectionDisabled(true)
+        .textInputAutocapitalization(.never)
+        .onAppear {
+            if showHiddenProfilesNotice && m.users.count > 1 {
+                alert = .hiddenProfilesNotice
+            }
+        }
         .confirmationDialog("Delete chat profile?", isPresented: $showDeleteConfirmation, titleVisibility: .visible) {
             deleteModeButton("Profile and server connections", true)
             deleteModeButton("Local profile data only", false)
         }
+        .sheet(item: $selectedUser) { user in
+            HiddenProfileView(user: user, profileHidden: $profileHidden)
+        }
+        .onChange(of: profileHidden) { _ in
+            DispatchQueue.main.asyncAfter(deadline: .now() + 10) {
+                withAnimation { profileHidden = false }
+            }
+        }
         .alert(item: $alert) { alert in
             switch alert {
-            case let .deleteUser(index, delSMPQueues):
+            case let .deleteUser(userInfo, delSMPQueues):
                 return Alert(
                     title: Text("Delete user profile?"),
                     message: Text("All chats and messages will be deleted - this cannot be undone!"),
                     primaryButton: .destructive(Text("Delete")) {
-                        removeUser(index, delSMPQueues)
+                        Task { await removeUser(userInfo, delSMPQueues) }
                     },
                     secondaryButton: .cancel()
                 )
+            case .cantDeleteLastUser:
+                return Alert(
+                    title: Text("Can't delete user profile!"),
+                    message: m.users.count > 1
+                            ? Text("There should be at least one visible user profile.")
+                            : Text("There should be at least use user profile.")
+                )
+            case .hiddenProfilesNotice:
+                return Alert(
+                    title: Text("Make profile private!"),
+                    message: Text("You can hide or mute a user profile - swipe it to the right.\nSimpleX Lock must be enabled."),
+                    primaryButton: .default(Text("Don't show again")) {
+                        showHiddenProfilesNotice = false
```

Comment:
maybe one time is enough? it's a bit irritating if I just click OK, which is bold. could be just a single button

Reply:
yes, but I thought it's important and people click OK without thinking. this way it'll at least will show again if they don't read it first time.